test/test_va_api_deriveimage.cpp

```diff
@@ -0,0 +1,137 @@
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT.
+ * IN NO EVENT SHALL PRECISION INSIGHT AND/OR ITS SUPPLIERS BE LIABLE FOR
+ * ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include "test_va_api_deriveimage.h"
+
+namespace VAAPI {
+// The following tests will operate on supported profile/entrypoint
+// combinations that the driver does support, there is no real need
+// to report SKIPPED tests as those cases are considered on the
+// get_create_config use cases and properly handled.
+
+VAAPIDeriveImage::VAAPIDeriveImage()
+{
+    m_vaDisplay = doInitialize();
+}
+
+VAAPIDeriveImage::~VAAPIDeriveImage() { doTerminate(); }
+
+
+TEST_P(VAAPIDeriveImage, DeriveImageWithAttributes)
+{
+
+    VAProfile currentProfile = ::testing::get<0>(GetParam());
+    VAEntrypoint currentEntrypoint = ::testing::get<1>(GetParam());
+    std::pair<uint32_t, uint32_t> currentResolution = ::testing::get<2>(GetParam());
+
+    doGetMaxValues();
+
+    doQueryConfigProfiles();
+
+    if (doFindProfileInList(currentProfile)) {
+
+	doQueryConfigEntrypoints(currentProfile);
+	if (doFindEntrypointInList(currentEntrypoint)) {
+	    // profile and entrypoint are supported
+
+	    doFillConfigAttribList();
+
+	    doGetConfigAttributes(currentProfile, currentEntrypoint);
+
+	    doQuerySurfacesWithConfigAttribs(currentProfile, currentEntrypoint);
+
+	    doCreateSurfaces(currentProfile, currentEntrypoint,
+                                           currentResolution);
+
+            doDeriveImage();
+	    doMapBuffer();
+	    doUnMapBuffer();
+	    doDestroyImage();
+	}
+    }
+}
+
+TEST_P(VAAPIDeriveImage, DeriveImageNoConfigAttributes)
+{
+    VAProfile currentProfile = ::testing::get<0>(GetParam());
+    VAEntrypoint currentEntrypoint = ::testing::get<1>(GetParam());
+    std::pair<uint32_t, uint32_t> currentResolution = ::testing::get<2>(GetParam());
+
+    doGetMaxValues();
+
+    doQueryConfigProfiles();
+
+    if (doFindProfileInList(currentProfile)) {
+
+        doQueryConfigEntrypoints(currentProfile);
+        if (doFindEntrypointInList(currentEntrypoint)) {
+            // profile and entrypoint are supported
+
+            doQuerySurfacesNoConfigAttribs(currentProfile,
+                                                   currentEntrypoint);
+
+            doCreateSurfaces(currentProfile, currentEntrypoint,
+                                           currentResolution);
+
+	   doDeriveImage();
+           doMapBuffer();
+           doUnMapBuffer();
+	   doDestroyImage();
+        }
+    }
+}
+
+TEST_P(VAAPIDeriveImage, DeriveImageNoAttributes)
+{
+    VAProfile currentProfile = ::testing::get<0>(GetParam());
+    VAEntrypoint currentEntrypoint = ::testing::get<1>(GetParam());
+    std::pair<uint32_t, uint32_t> currentResolution = ::testing::get<2>(GetParam());
```

Comment:
make this a const reference variable to avoid unnecessary copies.

Reply:
Fixed it.